exporter/opensearchexporter/factory.go

```diff
@@ -0,0 +1,13 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package opensearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opensearchexporter"
+
+import (
+	"go.opentelemetry.io/collector/exporter"
+)
+
+// NewFactory creates a factory for OpenSearch exporter.
+func NewFactory() exporter.Factory {
+	return nil
```

Comment:
This needs to return an actual component factory.  Even if the handler functions drop data on the floor, it should be possible after this PR to include the component in a build and wire it into a pipeline.

Reply:
Made change to add component factory.